src/test/assets_tests.cpp

```diff
@@ -200,900 +204,459 @@ CreateMintAssetTx(const CTxMemPool &mempool, SimpleUTXOMap &utxos, const CKey &c
-CTxOut out(1 * COIN, scriptPubKey);
-tx2.vout.
-push_back(out);
-//change
-CScript scriptPubKey2 = GetScriptForDestination(coinbaseKey.GetPubKey().GetID());
-CAssetTransfer assetTransfer2(assetId, 87.88 * COIN);
-assetTransfer2.
-BuildAssetTransaction(scriptPubKey2);
-CTxOut out2(0, scriptPubKey2);
-tx2.vout.
-push_back(out2);
-
-tx2.vin.
-push_back(CTxIn(COutPoint(tx.GetHash(), 0))
-);
-
-FundTransaction(tx2, utxos, GetScriptForDestination(coinbaseKey.GetPubKey().GetID()),
-1 * COIN, coinbaseKey);
-SignTransaction(*m_node
-.mempool, tx2, coinbaseKey);
-
-auto block = std::make_shared<CBlock>(CreateBlock({tx2}, coinbaseKey));
-EnsureChainman(m_node)
-.
-
-ProcessNewBlock(Params(), block,
+    CKey sporkKey;
+    sporkKey.MakeNewKey(false);
+    sporkManager.SetSporkAddress(EncodeDestination(sporkKey.GetPubKey().GetID()));
+    sporkManager.SetPrivKey(EncodeSecret(sporkKey));
+    sporkManager.UpdateSpork(SPORK_22_SPECIAL_TX_FEE, 2560, *m_node.connman);
 
-true, nullptr);
+    auto utxos = BuildSimpleUtxoMap(m_coinbase_txns);
 
-BOOST_ASSERT (::ChainActive()
+    auto tx = CreateNewAssetTx(*m_node.mempool, utxos, coinbaseKey, "Test Asset", true, false, 0, 8, 1000);
+    std::vector<CMutableTransaction> txns = {tx};
 
-.
+    int nHeight = ::ChainActive().Height();
 
-Height()
+    // Mining a block with a asset create transaction
+    {
+        auto block = std::make_shared<CBlock>(CreateBlock(txns, coinbaseKey));
+        EnsureChainman(m_node).ProcessNewBlock(Params(), block, true, nullptr);
 
-== nHeight + 2);
-BOOST_ASSERT(block
-->
-
-GetHash()
-
-!= ::ChainActive().Tip()->GetBlockHash());
-}
-
-//create a unique asset
-tx = CreateNewAssetTx(*m_node.mempool, utxos, coinbaseKey, "Unique Asset", false, true, 0, 0, 10);
-txns = {tx};
-
-nHeight = ::ChainActive().Height();
-
-// Mining a block with a asset create transaction
-{
-auto block = std::make_shared<CBlock>(CreateBlock(txns, coinbaseKey));
-EnsureChainman(m_node)
-.
+        BOOST_ASSERT(::ChainActive().Height() == nHeight + 1);
+        BOOST_ASSERT(block->GetHash() == ::ChainActive().Tip()->GetBlockHash());
+    }
 
-ProcessNewBlock(Params(), block,
+    std::string assetId = tx.GetHash().ToString();
+    tx = CreateMintAssetTx(*m_node.mempool, utxos, coinbaseKey, assetId);
 
-true, nullptr);
+    {
+        auto block = std::make_shared<CBlock>(CreateBlock({tx}, coinbaseKey));
+        EnsureChainman(m_node).ProcessNewBlock(Params(), block, true, nullptr);
 
-BOOST_ASSERT (::ChainActive()
+        BOOST_ASSERT(::ChainActive().Height() == nHeight + 2);
+        BOOST_ASSERT(block->GetHash() == ::ChainActive().Tip()->GetBlockHash());
+    }
 
-.
+    CAssetMetaData asset;
+    BOOST_ASSERT(passetsCache->GetAssetMetaData(assetId, asset));
 
-Height()
+    BOOST_ASSERT(asset.circulatingSupply == 1000 * COIN);
 
-== nHeight + 1);
-BOOST_ASSERT(block
-->
+    // Allow TX index to catch up with the block index.
+    g_txindex->BlockUntilSyncedToCurrentChain();
```

Comment:
This is what was missing.  Since we just created the transaction we need to reference, we need to wait for the index to be written before using it.